test/gtest/ucp/test_ucp_am.cc

```diff
@@ -603,9 +637,16 @@ class test_ucp_am_nbx_rndv: public test_ucp_am_nbx {
         EXPECT_TRUE(rx_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_RNDV);
         EXPECT_FALSE(rx_param->recv_attr & UCP_AM_RECV_ATTR_FLAG_DATA);
 
-        m_rx_buf.resize(length, 'u');
+        if (header_length != 0) {
+            EXPECT_EQ(std::string::npos,
+                      std::string((const char*)header,
+                                  header_length).find_first_not_of('h'));
```

Comment:
maybe create random pattern and compare to it?

Reply:
would need to save ptr to header somewhere then for comparing, i'd keep the test simpler instead